ARCs/arc-0056.md

```diff
@@ -0,0 +1,343 @@
+     * key. A key containing the human-readable name of the network MAY be
+     * included, but the corresponding genesis hash key MUST also be defined
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
+    };
+  };
+  /** Named structs use by the application */
+  structs: { [structName: StructName]: StructFields };
+  /** All of the methods that the contract implements */
+  methods: Method[];
+  state: {
+    /** Defines the values that should be used for GlobalNumUint, GlobalNumByteSlice, LocalNumUint, and LocalNumByteSlice when creating the application  */
+    schema: {
+      global: {
+        ints: number;
+        bytes: number;
+      };
+      local: {
+        ints: number;
+        bytes: number;
+      };
+    };
+    /** Mapping of human-readable names to StorageKey objects */
+    keys: {
+      global: { [name: string]: StorageKey };
+      local: { [name: string]: StorageKey };
+      box: { [name: string]: StorageKey };
+    };
+    /** Mapping of human-readable names to StorageMap objects */
+    maps: {
+      global: { [name: string]: StorageMap };
+      local: { [name: string]: StorageMap };
+      box: { [name: string]: StorageMap };
+    };
+  };
+  /** Supported bare actions for the contract. An action is a combination of call/create and an OnComplete */
+  bareActions: {
+    /** OnCompletes this method allows when appID === 0 */
+    create: ("NoOp" | "OptIn" | "DeleteApplication")[];
+    /** OnCompletes this method allows when appID !== 0 */
+    call: (
+      | "NoOp"
+      | "OptIn"
+      | "CloseOut"
+      | "ClearState"
+      | "UpdateApplication"
+      | "DeleteApplication"
+    )[];
+  };
+  /** Information about the TEAL */
+  sourceInfo?: SourceInfo[];
+  /** The pre-compiled TEAL that may contain template variables. MUST be omitted if included as part of ARC23, but otherwise MUST be defined. */
+  source?: {
+    /** The approval program */
+    approval: string;
+    /** The clear program */
+    clear: string;
+  };
+  /** ARC-28 events that MAY be emitted by this contract */
+  events?: Array<Event>;
+  /** A mapping of template variable names as they appear in the teal (not including TMPL_ prefix) and their respecive types */
+  templateVariables?: {
+    [name: string]: ABIType | AVMBytes | StructName;
+  };
+}
+```
+
+### Method Interface
+
+Every method in the contract is described via a `Method` interface. This interface is an extension of the one defined in [ARC-4](./arc-0004.md).
+
+```ts
+/** Describes a method in the contract. This interface is an extension of the interface described in ARC-4 */
+interface Method {
+  /** The name of the method */
+  name: string;
+  /** Optional, user-friendly description for the method */
+  desc?: string;
+  /** The arguments of the method, in order */
+  args: Array<{
+    /** The type of the argument */
+    type: ABIType;
```

Comment:
What should the value of `type` be if `struct` is also defined?

Reply:
Ah, the equivalent ARC-4 tuple type string I guess?

Reply:
Yes just the ABI type for backwards compat with ARC4. ARC56 parsers read `struct ?? type` to determine the ARC56 type to use

Reply:
ah, maybe this is the answer to the question I just asked. It wasn't immediately obvious to me that I should do that. We should probably indicate it in the description comments at the very least to make it more obvious for implementers?

Reply:
Same detail for returns...